lib/repositories/firebase/transaction_repository.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:async';
+import 'package:cloud_firestore/cloud_firestore.dart' show Firestore;
+
+import 'package:pispapp/models/transaction.dart';
+import 'package:pispapp/repositories/interfaces/i_transaction_repository.dart';
+
+typedef TransactionHandler = void Function(Transaction);
+
+class TransactionRepository implements ITransactionRepository {
+  final _colRef = Firestore.instance.collection('transactions');
+
+  @override
+  Future<List<Transaction>> getTransactions(String userId) async {
+    final transactionDocs =
+        await _colRef.where('userId', isEqualTo: userId).getDocuments();
```

Comment:
I don't think you need an "await" here, you can just `.then` and map it since the return type is a future anyways.

Reply:
oh right, will fix this!

Reply:
fixed